### ISSUE TITLE
Decide on correct OSS License and implement throughout

### ISSUE THREAD

**Issue by thclark** [octue/octue-sdk-python#67]
## Licensing

We have several libraries now under OSS but havent formally decided what license to use.

@AndyClifton suggested some variant (?) of hte BSD license was superiori to MIT (which we currently use) because of <reason>.

Andy, can you remember the reason and clarify that variant?

Once we have decided which to use, we'll apply throughout to all repos public on github.com/octue including this one.

**Comment by AndyClifton** [comment]
Hi... I think the BSD was preferred because of the option of the 4-clause version. [See stackoverflow for more](https://opensource.stackexchange.com/questions/217/what-are-the-essential-differences-between-the-bsd-and-mit-licences).

> All advertising materials mentioning features or use of this software must display the following acknowledgement: This product includes software developed by the [organization].

But, there's a risk that the advertising requirement would be too much for some potential users.

See https://choosealicense.com/appendix/ for an exhaustive list.

**Comment by AndyClifton** [comment]
tl, dr: choose a license that fits with the community you are working with. If your community knows MIT and uses it, use MIT.

**Comment by cortadocodes** [comment]
@thclark given what @AndyClifton has said, is there a canonical licence of the MIT or BSD type that we can add to the `.github` repo so it applies as default in all repos without their own licence file?

**Comment by thclark** [comment]
> @thclark given what @AndyClifton has said, is there a canonical licence of the MIT or BSD type that we can add to the `.github` repo so it applies as default in all repos without their own licence file?

No, I definitely don't want to apply a default license in an automated way, I want that bound and committed into the source code. I also want to make that decision on a case-by-case basis, but organisationally adopt one as a defacto license.

I'll review the BSD licenses again and have a look at the 4-clause license. It might be that we apply that then offer an additional license to customers waiving them from that clause. It does protect somewhat against the problem that the docker and k8s folks had where suddenly their OSS work was being used all over the place to make money for the likes of GCP and Azure without any onward revenue being generated.

Assignment over to just me.

**Comment by thclark** [comment]
The update section of [this post](https://wonko.com/post/jsmin-isnt-welcome-on-google-code) is hilarious :rofl:

**Comment by thclark** [comment]
Worth also considering BSL https://blog.sentry.io/2019/11/06/relicensing-sentry

**Comment by AndyClifton** [comment]
> Worth also considering BSL https://blog.sentry.io/2019/11/06/relicensing-sentry
> https://www.cockroachlabs.com/blog/oss-relicensing-cockroachdb/

The sentry discussion is entirely relevant in this case.

**Comment by thclark** [comment]
Additional resource is the Commons Clause which can be applied on top of an existing OSS license. https://commonsclause.com/ which could be appropriate as a modifier
